### PR TITLE
Reworking `help aliases`

### PR DESCRIPTION
**Pull request by klementievdmitry**
# Description

I reworked the `help aliases` command a bit. It now collects old and new aliases

# User-Facing Changes

* previously aliases, were not listed in the `help aliases`:
```nu
$ help aliases | where name == "bar"
╭────────────╮
│ empty list │
╰────────────╯
$ alias bar = echo "this is bar"
$ help aliases | where name == "bar"
╭────────────╮
│ empty list │
╰────────────╯
```

* now the aliases, will be listed in `help aliases` and `$nu.scope.aliases`:
```nu
$ help aliases | where name == "bar"
╭────────────╮
│ empty list │
╰────────────╯
$ $nu.scope.aliases | where name == "bar"
╭────────────╮
│ empty list │
╰────────────╯
$ alias bar = echo "this is bar"
$ help aliases | where name == "bar"
╭───┬──────┬────────────────────┬───────╮
│ # │ name │     expansion      │ usage │
├───┼──────┼────────────────────┼───────┤
│ 0 │ bar  │ echo "this is bar" │       │
╰───┴──────┴────────────────────┴───────╯
$ $nu.scope.aliases | where name == "bar"
╭───┬──────┬────────────────────┬───────╮
│ # │ name │     expansion      │ usage │
├───┼──────┼────────────────────┼───────┤
│ 0 │ bar  │ echo "this is bar" │       │
╰───┴──────┴────────────────────┴───────╯
```

# Tests + Formatting

Don't forget to add tests that cover your changes.

Make sure you've run and fixed any issues with these commands:

- `cargo fmt --all -- --check` to check standard code formatting (`cargo fmt --all` applies these changes)
- `cargo clippy --workspace -- -D warnings -D clippy::unwrap_used -A clippy::needless_collect` to check that you're using the standard code style
- `cargo test --workspace` to check that all tests pass

# After Submitting

If your PR had any user-facing changes, update [the documentation](https://github.com/nushell/nushell.github.io) after the PR is merged, if necessary. This will help us keep the docs up to date.

### CODE REVIEW

File: crates/nu-cmd-lang/src/core_commands/help_aliases.rs [hunk 1]
```diff
@@ -164,10 +164,53 @@ pub fn help_aliases(
...
     let mut scope_data = ScopeData::new(engine_state, stack);
     scope_data.populate_aliases();
+    aliases.append(&mut scope_data.collect_aliases(span));
+
+    // Getting all other aliases
+    let commands = engine_state.get_decls_sorted(false);
+
+    for (name_bytes, decl_id) in commands {
+        let decl = engine_state.get_decl(decl_id);
+        if format!("{:?}", decl.command_type()).to_lowercase() != "alias" {
```
Comment (rgwood) [inline crates/nu-cmd-lang/src/core_commands/help_aliases.rs:180]:
```suggestion
        if !decl.is_alias() {
```

**Review by rgwood — CHANGES_REQUESTED** [review]
Nice, thanks for fixing this up! Looks good overall, just 2 comments.

@kubouch, are you OK with this or will it conflict with the changes you're making to aliases?

**Comment by klementievdmitry** [comment]
What do you think about moving the implementation from `build_help_aliases` to `collect_aliases`? As I understand it, it will also affect `$nu.scope.aliases`.

**Comment by rgwood** [comment]
I think that's a good idea. Let's wait until @kubouch has time to take a quick look, he's been doing a lot of work in this area recently.

**Comment by kubouch** [comment]
Cool, looks good, thanks!

**Pull request merged**